integration-tests/tests/src/tests/observable.ts

```diff
@@ -0,0 +1,848 @@
+  collection.addListener(
+    createListenerStub(
+      handle,
+      ...expectedChangeSets.map((expectedChanges, c) => (_: Realm.Collection, changes: CollectionChangeSet) => {
+        expect(changes).deep.equals(expectedChanges, `Changeset #${c} didn't match`);
+      }),
+    ),
+  );
+  return handle;
+}
+
+function expectDictionaryNotifications(dictionary: Realm.Dictionary, expectedChangeSets: DictionaryChangeSet[]) {
+  const handle = createPromiseHandle();
+  dictionary.addListener(
+    createListenerStub(
+      handle,
+      ...expectedChangeSets.map((expectedChanges, c) => (_: Realm.Dictionary, changes: DictionaryChangeSet) => {
+        expect(changes).deep.equals(expectedChanges, `Changeset #${c} didn't match`);
+      }),
+    ),
+  );
+  return handle;
+}
+
+describe("Observable", () => {
+  // describe("App", () => {});
+
+  describe("Realm", () => {
+    openRealmBeforeEach({
+      schema: [
+        {
+          name: "Person",
+          properties: {
+            name: "string",
+          },
+        },
+      ],
+    });
+
+    it("is observable", function (this: RealmContext) {
+      expectObservableMethods(this.realm);
+    });
+
+    describe("change", () => {
+      it("calls listener", async function (this: RealmContext) {
+        const completion = expectRealmNotifications(this.realm, "change", [{}]);
+
+        this.realm.write(() => {
+          this.realm.create("Person", { name: "Alice" });
+        });
+
+        await completion;
+      });
+
+      it("removes listeners", async function (this: RealmContext) {
+        const handle = createPromiseHandle();
+
+        const listener = createListenerStub(handle, () => {
+          this.realm.removeListener("change", listener);
+          setImmediate(() => {
+            this.realm.write(() => {
+              this.realm.create("Person", { name: "Bob" });
+            });
+          });
+        });
+
+        this.realm.addListener("change", listener);
+
+        this.realm.write(() => {
+          this.realm.create("Person", { name: "Alice" });
+        });
+
+        await handle;
+      });
+
+      it("removes all listeners", async function (this: RealmContext) {
+        const handle = createPromiseHandle();
+
+        const listener = createListenerStub(handle, () => {
+          this.realm.removeAllListeners("change");
+          setImmediate(() => {
+            this.realm.write(() => {
+              this.realm.create("Person", { name: "Bob" });
+            });
+          });
+        });
+
+        this.realm.addListener("change", listener);
+
+        this.realm.write(() => {
+          this.realm.create("Person", { name: "Alice" });
+        });
+
+        await handle;
+      });
+    });
+
+    describe("beforenotify", () => {
+      // Skipping on React Native because the callback is called one too many times
+      it.skipIf(environment.reactNative, "calls listener", async function (this: RealmContext) {
```

Comment:
We were already skipping this test on React Native:

https://github.com/realm/realm-js/blob/ca6abdd242a838e3d9c70a99e3a0fecf110e86e0/integration-tests/tests/src/tests/listeners.ts#L44-L45

I still don't know why this is happening and it would make sense for us to look into it at some point. I've created https://github.com/realm/realm-js/issues/6298 to track this.

Reply:
Does it happens both on Android and iOS? I mean, the event loop is different and I wonder if it is related to core's scheduler or our code?

Reply:
Yes, it happens on Android and iOS.